### PR TITLE
Enhancement #7980: Banned website UI part 3

### LINKED ISSUE #7980: Banned websites UI changes | Part 3 | License updates
Part 3 of #7973

Update license info display on dashboard to reflect revoked/banned status and show green check mark if everything is working fine with updates.
I think this is not only related to revoked sites, for expired users also we don't show updates, to be validated with @wp-media/engineering-wpr-addons-team and @DahmaniAdame on which cases other than banning a website that stops updating the plugin.

**As a reseller, I want to update WPR status to have plugin updates state**
https://www.notion.so/wpmedia/Reseller-banned-users-UI-changes-225ed22a22f080b7b803e403ec637f6f?source=copy_link#225ed22a22f08028b308e2f1a2dd5df7
**AC:**

- status isnot displayed to whitelabel?
- if reseller website isnot banned, only green check will be displayed beside plugin updates status

### PR DESCRIPTION
**Pull request by Miraeld**
# Description
This pull request enhances the plugin's license and update management by introducing new logic to determine if plugin updates are available, the reasons for blocking updates, and whether a site is banned. It also updates the admin dashboard to display this information and adds comprehensive unit tests and fixtures for the new functionality.

Fixes #7980 
## Type of change

- [x] New feature (non-breaking change which adds functionality).

### What was tested

**Automated:**
- Unit tests for `can_update_plugin()` and `get_update_blocked_reason()` methods (11 test cases covering all scenarios)

**Manual:**
- Dashboard UI displays green checkmark for valid licenses
- Dashboard UI displays error messages for banned/expired licenses
- Plugin Updates section hidden for whitelabel accounts

### How to test

1. Run unit tests: `composer run test-unit -- tests/Unit/inc/Engine/License/API/User/`
2. To test UI manually, add this filter to `wp-content/mu-plugins/test.php`:
   ```php
   add_filter( 'transient_wp_rocket_customer_data', function ( $customer_data ) {
       if ( ! is_object( $customer_data ) ) {
           return $customer_data;
       }
       // For expired: 
       $customer_data->licence_expiration = strtotime( 'last year' );
       $customer_data->licence->is_banned = false;
       
       // For banned:
       // $customer_data->licence->is_banned = true;
       // $customer_data->licence->plugin_updates_ban_reason = 'BANNED_WEBSITE';
       
       return $customer_data;
   } );

## Technical description

### Documentation

This pull request enhances the plugin's license and update management by introducing new logic to determine if plugin updates are available, the reasons for blocking updates, and whether a site is banned. It also updates the admin dashboard to display this information and adds comprehensive unit tests and fixtures for the new functionality.

**License and Update Management Enhancements:**

- Added new methods to the `User` class (`inc/Engine/License/API/User.php`) to check if a website is banned, retrieve the ban reason, determine if plugin updates are allowed, and get the reason why updates are blocked. These methods ensure more precise control and messaging around plugin updates and license status.

- Updated the `customer_data()` method in `Settings/Page.php` to include new fields: `can_update_plugin` and `update_blocked_reason`, leveraging the new `User` methods for more granular update status reporting.

**Admin Dashboard UI Update:**

- Modified the dashboard view (`views/settings/page-sections/dashboard.php`) to display plugin update status and, if updates are blocked, show the reason to the user, improving transparency and user experience.

**Testing Improvements:**

- Added unit tests for the new `User` methods (`is_banned`, `ban_reason`, `can_update_plugin`, `get_update_blocked_reason`) to ensure correct behavior under various scenarios. [[1]](diffhunk://#diff-82888b2f5df623eb919c13e6d99e20b78fe75b268e5b98293c1a8b9767424826R1-R25) [[2]](diffhunk://#diff-4b34758a460ae215d721cf67bec1858cdeacd5bbffebf70aca38d2417911b1bcR1-R25) [[3]](diffhunk://#diff-cc4a362f9ee38fa967f76c43a34e5ab025eee8d46a0fc69c470bfc44c973f188R1-R25) [[4]](diffhunk://#diff-b3e301901208307828593ec6cab558984676482e2ac9a52b105de133dedd3630R1-R27)
- Introduced corresponding test fixtures to cover different edge cases and data combinations for these methods. [[1]](diffhunk://#diff-22ce8735650257e11cbd9022695cc1f893889d05a9c6984d9ce6c3267af20057R1-R57) [[2]](diffhunk://#diff-9456cf25aeb7b98be82d22f8fd16a62692500e7f42a7359a1548c11fafeaeb1eR1-R30) [[3]](diffhunk://#diff-bdc2f3dadd47686787436fc8a48918a42bb5542856a9035eb8e8e9919c3244e8R1-R46) [[4]](diffhunk://#diff-97533c0bf66cf25df754f81e832839da4bcf27749ba5caff0f07f1761e71fc2aR1-R57)

**Dependency Update:**

- Added the missing import for the `User` class in `Settings/Page.php` to support the new logic.

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style

- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [x] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

### CODE REVIEW

**Comment by Miraeld** [comment]
Screenshot 2026-01-20 at 07 00 59
Screenshot 2026-01-20 at 06 54 17

File: inc/Engine/Admin/Settings/Page.php [hunk 1]
```diff
@@ -285,32 +286,39 @@ public function async_wistia_script( $tag, $handle ) {
...
+			'license_type'          => __( 'Unavailable', 'rocket' ),
+			'license_expiration'    => __( 'Unavailable', 'rocket' ),
+			'license_class'         => 'wpr-isInvalid',
+			'is_from_one_dot_com'   => false,
+			'can_update_plugin'     => false,
+			'update_blocked_reason' => '',
 		];
 
-		$data['license_type'] = rocket_get_license_type( $user );
+		$data['license_type'] = rocket_get_license_type( $user_data );
```
Comment (jeawhanlee) [inline inc/Engine/Admin/Settings/Page.php:300]: We also need to cater for revoked license, check [here](https://www.notion.so/wpmedia/Reseller-banned-users-UI-changes-225ed22a22f080b7b803e403ec637f6f?source=copy_link#2bfed22a22f080beb342ee0e204eb8ce)

File: inc/Engine/Admin/Settings/Page.php [hunk 2]
```diff
@@ -285,32 +286,39 @@ public function async_wistia_script( $tag, $handle ) {
...
-		if ( ! empty( $user->licence_expiration ) ) {
-			$data['license_class'] = time() < $user->licence_expiration ? 'wpr-isValid' : 'wpr-isInvalid';
+		if ( ! empty( $user_data->licence_expiration ) ) {
+			$data['license_class'] = time() < $user_data->licence_expiration ? 'wpr-isValid' : 'wpr-isInvalid';
 		}
 
-		if ( ! empty( $user->licence_expiration ) ) {
-			$data['license_expiration'] = date_i18n( get_option( 'date_format' ), (int) $user->licence_expiration );
+		if ( ! empty( $user_data->licence_expiration ) ) {
+			$data['license_expiration'] = date_i18n( get_option( 'date_format' ), (int) $user_data->licence_expiration );
```
Comment (jeawhanlee) [inline inc/Engine/Admin/Settings/Page.php:313]: Here as well

File: inc/Engine/License/API/User.php [hunk 3]
```diff
@@ -528,4 +528,68 @@ protected function get_rocket_insights_data( string $sku ) {
...
+			return __( 'There was an error updating the plugin because your website has been suspended.', 'rocket' );
+		}
+
+		// Check if license is expired.
+		if ( $this->is_license_expired() ) {
+			return __( 'There was an error updating the plugin because your license has expired.', 'rocket' );
+		}
+
+		return '';
+	}
```
Comment (jeawhanlee) [inline inc/Engine/License/API/User.php:596]: I think we might have to change the logic a bit here based on the AC [here](https://www.notion.so/Reseller-banned-users-UI-changes-225ed22a22f080b7b803e403ec637f6f?d=2aaed22a22f080d98756001ce647a6b0&source=copy_link#225ed22a22f080fd9db4e37c110a99f2) As there was no cater for expired license. From what I understand, if there's no reason, we'll leave it as "There was an error updating the plugin" and there was a reason, we go ahead to add the "because" followed by the reason

**Review by jeawhanlee — CHANGES_REQUESTED** [review]

**Comment by wordpressfan** [comment]
I pushed some commits to fix this based on the slack discussion here:
https://group-onecom.slack.com/archives/C08EFCYRQ2X/p1769001340267579?thread_ts=1768902278.704059&cid=C08EFCYRQ2X

And the review comments from @jeawhanlee

File: inc/Engine/Admin/Settings/Page.php [hunk 4]
```diff
@@ -285,32 +286,41 @@ public function async_wistia_script( $tag, $handle ) {
...
+			$data['license_class'] = ( time() < $user_data->licence_expiration && ! $user->is_banned() ) ? 'wpr-isValid' : 'wpr-isInvalid';
 		}
 
-		if ( ! empty( $user->licence_expiration ) ) {
-			$data['license_expiration'] = date_i18n( get_option( 'date_format' ), (int) $user->licence_expiration );
+		if ( $user->is_banned() ) {
+			$data['license_expiration'] = __( 'Ended', 'rocket' );
+		} elseif ( ! empty( $user_data->licence_expiration ) ) {
+			$data['license_expiration'] = date_i18n( get_option( 'date_format' ), (int) $user_data->licence_expiration );
 		}
```
Comment (jeawhanlee) [inline inc/Engine/Admin/Settings/Page.php:314]:
```suggestion
		if ( ! empty( $user_data->licence_expiration ) ) {
			$data['license_class'] = ( time() < $user_data->licence_expiration && ! $user->is_banned() ) ? 'wpr-isValid' : 'wpr-isInvalid';
			
		$data['license_expiration'] = date_i18n( get_option( 'date_format' ), (int) $user_data->licence_expiration );
}

		if ( $user->is_banned() ) {
			$data['license_expiration'] = __( 'Ended', 'rocket' );
		} 
```

File: inc/Engine/License/API/User.php [hunk 5]
```diff
@@ -539,11 +539,73 @@ public function is_banned() {
...
+					__( 'There was an error updating the plugin because %s', 'rocket' ),
+					$reason_text
+				);
+			}
+			return __( 'There was an error updating the plugin.', 'rocket' );
+		}
+
+		// Check if license is expired.
+		if ( $this->is_license_expired() ) {
+			return __( 'There was an error updating the plugin.', 'rocket' );
```
Comment (jeawhanlee) [inline inc/Engine/License/API/User.php]: Instead of returning the same text each time, can't we define the text once and return it and if we need to return the reason, we can concatenate the variable:
$string .= sprintf( __( 'because %s', 'rocket' ), $reason );

File: inc/Engine/License/API/User.php [hunk 6]
```diff
@@ -539,11 +539,73 @@ public function is_banned() {
...
+		return '';
+	}
+
+	/**
+	 * Convert ban reason code to human-readable text.
+	 *
+	 * @param string $ban_reason Ban reason code.
+	 * @return string
+	 */
+	private function get_ban_reason_text( $ban_reason ) {
```
Comment (jeawhanlee) [inline inc/Engine/License/API/User.php]: Shouldn't method name be generic and not limited to ban?

File: views/settings/page-sections/dashboard.php [hunk 7]
```diff
@@ -112,6 +112,16 @@
 							<span class="wpr-title3"><?php esc_html_e( 'Expiration Date', 'rocket' ); ?></span>
 							<span class="wpr-infoAccount <?php echo esc_attr( $data['customer_data']['license_class'] ); ?>" id="wpr-expiration-data"><?php echo esc_html( $data['customer_data']['license_expiration'] ); ?></span>
 						</p>
+						<?php if ( ! defined( 'WP_ROCKET_WHITE_LABEL_ACCOUNT' ) || ! WP_ROCKET_WHITE_LABEL_ACCOUNT ) : ?>
+						<p>
+							<span class="wpr-title3"><?php esc_html_e( 'Plugin Updates', 'rocket' ); ?></span>
+							<?php if ( ! empty( $data['customer_data']['can_update_plugin'] ) ) : ?>
+								<span class="wpr-infoAccount wpr-isValid wpr-icon-check" id="wpr-plugin-updates-data"></span>
+							<?php else : ?>
+								<span class="wpr-infoAccount wpr-isInvalid" id="wpr-plugin-updates-data"><?php echo esc_html( $data['customer_data']['update_blocked_reason'] ); ?></span>
```
Comment (jeawhanlee) [inline views/settings/page-sections/dashboard.php:121]: @wp-media/product Do we need the exclamation icon to be the exact same one on notion [here](https://www.notion.so/wpmedia/Reseller-banned-users-UI-changes-225ed22a22f080b7b803e403ec637f6f?source=copy_link#2bfed22a22f080beb342ee0e204eb8ce)

**Review by jeawhanlee — CHANGES_REQUESTED** [review]

**Comment by wordpressfan** [comment]
I believe we are fine here, let's merge this so @wp-media/qa-team can validate and move forward.
